### DISCUSSION TITLE
Doctrine queries behave differently when executed in the twig or in the controller.

### DISCUSSION THREAD

**Discussion by VincentLanglet** [symfony/symfony#49510] [Q&A]
Hi, I’m using a [Doctrine SQLFilter](https://www.doctrine-project.org/projects/doctrine-orm/en/2.14/reference/filters.html) (enabled by default) and having a weird behavior with it.

I have a KernelRequest listener which is calling setParameterList on this SQLFilter to pass some config.
Then, when I use the entity getter in my controller,
```
$projects = $organization->getProject();
```
I get a Lazyloaded Collection which have the following issue:
If I call count on it inside the controller `$projects->count()`, I get 4 ; which is the expected result.
If I pass the value to the twig and call count on it inside the view `project|length`, I get 2
If of course, if I call count on both side, I get 4, because the twig does not retrigger a sql query.

So it seems like the param passed to the SQLFilter inside the KernelRequest listener is correctly used when the SQL query is triggered (by the count call) inside the controller but not inside the twig. 
Moreover, I only get this different behavior between controller and twig queries in PROD.

Any idea 
- Why i have a different behavior between controller and twig
- Why I have a different behavior between prod and dev

I have trouble to find if it’s a doctrine issue, a Symfony listener/request issue, a twig issue, ...

**Comment by stof** [comment]
The Twig `length` filter calls `count()` for Countable objects: https://github.com/twigphp/Twig/blob/78c18ad58e48d3fff2f7812bb60b6176ee52689a/src/Extension/CoreExtension.php#L1118-L1141

So it looks weird to have differences between a count call done in the controller or in Twig with only that difference.

anyway, without a reproducer, there is not much we can do here.

**Reply by VincentLanglet** [reply]
Sorry, I found my issue.
it was because in my twig I had a twig extensions which was playing with doctrine filters and the following issue https://github.com/doctrine/orm/issues/10536
